### PR TITLE
Feature - Send Mail via Model Events

### PR DESCRIPTION
**Pull request by adammparker**
Model Events (Index)
image

Model Events (Detail)
image

Model Events (Form)
image

### CODE REVIEW

**Comment by adammparker** [comment]
@jrseliga Do you think we should add a check to the `bootMailable` method to make sure the model is still in the `eventables` key in `nova_mail` config?

My thought there being someone might add `User::class` and then add some model events but then if they want to completely disable model based e-mail they might think they can just remove the model from the `eventables` key in the config, but that really just controls if it shows up in the dropdown.

File: config/nova_mail.php [hunk 1]
```diff
@@ -49,5 +49,22 @@
...
+    | Here you can specify which classes can have dynamic
+    | listeners for sending Nova Mail Templates.
+    |
+    | Ex: [User::class]
+    |
+    | Note: These classes must use the Mailable trait and
+    |       implement the abstract method getEmailField.
+    ]
+    */
+    'eventables' => [],
```
Comment (jrseliga) [inline config/nova_mail.php]: @adammparker wondering if we should either `filter()` out any that don't use the trait or ship and throw an `Exception` to provide developers with a bit more information

File: README.md [hunk 2]
```diff
@@ -92,6 +94,16 @@ class User extends Resource
 
 Now you can send emails from the action called "Send Mail" on your resource!
 
+### Trigger Mail on Model Events
+
+You can create a `MailTemplate` with a Model Event that monitors an 'active' column on your `User` model that when toggled sends an email to the user letting them know the current status of their account:
```
Comment (jrseliga) [inline README.md]: @adammparker
Maybe something like this instead?
> A `MailTemplate` can be configured to respond to Model events or a value change of a specified column. For example, a mail template informing your users of their account status could be sent when the `active` column on your `User` model is updated:

**Comment by jrseliga** [comment]
@adammparker in regards to https://github.com/kirschbaum-development/nova-mail/pull/21#issuecomment-526567413, yeah I think we should filter these out, good catch

File: src/Models/NovaSentMail.php [hunk 3]
```diff
@@ -83,4 +67,18 @@ public function setMailTemplateAttribute($value)
...
+    /**
+     * The "booting" method of the model.
+     *
+     * @return void
+     */
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::creating(function ($mail) {
```
Comment (belisarh) [inline src/Models/NovaSentMail.php:80]: What if there is no authenticated user and the event was triggered by an update outside the browser request cycle?

**Pull request merged**